### PR TITLE
remove from transport protocol from crate http

### PR DESCRIPTION
**Pull request by kovrus**

### CODE REVIEW

**Comment by kovrus** [comment]
@bogensberger please pre review

**Comment by bogensberger** [comment]
lgtm

**Comment by kovrus** [comment]
@mfussenegger please review

File: sql/src/main/java/io/crate/rest/action/RestResultSetReceiver.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    @Override
+    public void addListener(CompletionListener listener) {
+        this.completionListener = CompletionMultiListener.merge(this.completionListener, listener);
+    }
+
+    @Override
+    public void setNextRow(Row row) {
+        rows.add(row.materialize());
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/rest/action/RestResultSetReceiver.java]: Wouldn't it be possible to write onto the XContentBuilder directly here? 
By doing that we could get rid of the `.materialize` calls

Comment (chaudum) [inline sql/src/main/java/io/crate/rest/action/RestResultSetReceiver.java]: @mfussenegger I've pushed a fixup where the xcontent is written directly

**Comment by chaudum** [comment]
@mfussenegger did rebase and used `BaseResultReceiver`

File: sql/src/main/java/io/crate/rest/action/RestBulkRowCountReceiver.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        this.resultIdx = resultIdx;
+    }
+
+    @Override
+    public void setNextRow(Row row) {
+        rowCount = ((long) row.get(0));
+    }
+
+    @Override
+    public void batchFinished() {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/rest/action/RestBulkRowCountReceiver.java]: override is not required

File: sql/src/main/java/io/crate/rest/action/RestResultSetReceiver.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    public void setNextRow(Row row) {
+        try {
+            builder.addRow(row, outputFields.size());
+        } catch (IOException e) {
+            fail(e);
+        }
+    }
+
+    @Override
+    public void batchFinished() {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/rest/action/RestResultSetReceiver.java]: override is not required

File: sql/src/main/java/io/crate/rest/action/RestRowCountReceiver.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        builder = tmpBuilder;
+    }
+
+    @Override
+    public void setNextRow(Row row) {
+        rowCount = (long) row.get(0);
+    }
+
+    @Override
+    public void batchFinished() {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/rest/action/RestRowCountReceiver.java]: override is not required

**Comment by mfussenegger** [comment]
see my comments regarding the unnecessary overrides, otherwise lgtm

**Comment by chaudum** [comment]
@mfussenegger ah. forgot to remove it again after reverting some changes I did already.

**Pull request merged**